src/libImaging/BcnEncode.c

```diff
@@ -0,0 +1,205 @@
+/*
+ * The Python Imaging Library
+ *
+ * encoder for DXTn-compressed data
+ *
+ * Format documentation:
+ *   https://web.archive.org/web/20170802060935/http://oss.sgi.com/projects/ogl-sample/registry/EXT/texture_compression_s3tc.txt
+ *
+ * The contents of this file are in the public domain (CC0)
+ * Full text of the CC0 license:
+ *   https://creativecommons.org/publicdomain/zero/1.0/
```

Comment:
if you want to release this into the public domain, or if you feel like this is based on https://github.com/python-pillow/Pillow/blob/main/src/libImaging/BcnDecode.c, this is fine - but just be aware it's not a standard part of our C files. See https://github.com/python-pillow/Pillow/blob/main/src/libImaging/BitDecode.c for example

Reply:
I see I'll change it. Encoder code is based on test code I wrote myself from scratch